time_unit/__init__.py

```
@@ -186,10 +235,10 @@ def __ge__(self, other):
         return int(self) >= int(other)
 
     def __int__(self):
-        return self.dt.year * 100000 + self.dt.month * 1000 + self.dt.day * 10 + self.kind.kind_int
+        return date_to_int(self.dt, self.kind.multiplier) + self.kind.kind_int
 
     def __repr__(self):
-        return f"Timeunit({self.kind.__qualname__}, {self.dt!r})"
+        return f"{self.__class__.__name}({self.kind.__qualname__}, {self.dt!r})"
```

Pylint (3.3.7)

[error] Class 'Timeunit' has no '__name' member

(E1101)